src/h5web/visualizations/nexus/hooks.ts

```diff
@@ -59,29 +54,29 @@ export function useNxData(group: HDF5Group, metadata: HDF5Metadata): NxData {
   const silxStyle = parseSilxStyleAttribute(group);
   const { axes_scale_type, signal_scale_type } = silxStyle;
 
-  const axisMapping = axesNames.map((name, i) => {
-    if (!name) {
+  const axisMapping = axesNames.map((axisName, i) => {
+    if (!axisName) {
       return undefined;
     }
 
-    const dataset = getLinkedEntity(name, group, metadata);
-    if (!dataset) {
-      return undefined;
-    }
-    assertDataset(dataset);
+    const axisDataset = getLinkedEntity(axisName, group, metadata);
+    assertDefined(axisDataset);
+    assertDataset(axisDataset);
 
-    const value = datasetValues && datasetValues[name];
-    assertOptionalArray<number>(value);
+    const axisValue = values[axisDataset.id];
+    assertOptionalArray<number>(axisValue);
 
     return {
-      value,
-      label: getDatasetLabel(dataset, name),
+      value: axisValue,
+      label: getDatasetLabel(axisDataset, axisName),
       scaleType: axes_scale_type && axes_scale_type[i],
     };
   });
 
-  const errors = datasetValues && datasetValues.errors;
-  assertOptionalArray<number>(errors);
+  const errorsLink = getLink('errors', group);
+  const errorsValue =
+    errorsLink && isReachable(errorsLink) && values[errorsLink.id];
+  assertOptionalArray<number>(errorsValue);
```

Comment:
Here is where I have to use `getLink`. This code will likely change again, as mentioned (and also because of #318).